### PR TITLE
Server-Side Rendering Vite

### PR DESCRIPTION
**Pull request by pepicrft**

### CODE REVIEW

File: packages/core/src/cli/fs.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+import fs from 'fs-extra'
 
 export async function exists(path: string): Promise<boolean> {
-    return fs.pathExists(path);
+  return fs.pathExists(path)
+}
+
+//@josemasar: DO NOT DELETE, used in server.ts
+//https://nodejs.org/api/fs.html#fsreadfilesyncpath-options
+// @pepibimur, I need to provide types for options to pass the encoding - help!?
```
Comment (josemasar) [inline packages/core/src/cli/fs.ts]: @pepicrft , here I need some help.

Comment (josemasar) [reply packages/core/src/cli/fs.ts]: Docs: https://nodejs.org/api/fs.html#fsreadfilesyncpath-options

Within options, I need to pass the 'encoding'.

File: packages/serve/src/cli/utilities/server.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    },
+  })
+
+  server.use(vite.middlewares)
+
+  server.use('*', async ({ originalUrl }, next) => {
+    const url = originalUrl
+
+    try {
+      const template = fs.readFile(
```
Comment (josemasar) [inline packages/serve/src/cli/utilities/server.ts]: This is where I use 'readFile' from 'fs'

File: packages/core/src/cli/fs.ts (hunk 1 above)
Comment (pepicrft) [reply packages/core/src/cli/fs.ts]: I'd recommend not passing anything for now, and add options as needed. So you'd change the method signature to:
```ts
export async function readFile(path: string): Promise<string> {}
```
I'd make it return a `string`, which is what we'll be reading most of the times

**Comment by josemasar** [comment]
@pepicrft 
image

**Comment by pepicrft** [comment]
I'm closing this one because it's got pretty stale. We can open a new one if needed off the most recent commit in `main`.

**Pull request closed without merging**